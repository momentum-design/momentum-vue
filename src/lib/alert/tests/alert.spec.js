import { mount } from '@vue/test-utils';
import Alert from '../index.vue';
import Button from '../../button/index.vue';
import Icon from '../../icon/index.vue';
import Vue from 'vue';

Vue.component(Button.name, Button);
Vue.component(Icon.name, Icon);

describe('Alert', () => {
  it('should match snapshot', () => {
    const wrapper = mount(Alert, {
      propsData: {
        show: true,
        dismissBtnProps: {
          ariaLabel: 'Close'
        }
      }
    });
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('should render one Alert', () => {
    const wrapper = mount(Alert, {
      propsData: {
        show: true,
        closable: false,
      }
    });

    expect(wrapper.findAll('.md-alert').length).toEqual(1);
  });

  it('should pass otherProps to wrapper', () => {
    const wrapper = mount(Alert, {
      propsData: {
        show: true,
        closable: false,
        id: 'testid',
      }
    });

    expect(wrapper.find('#testid').exists()).toEqual(true);
  });

  it('should pass class prop', () => {
    const wrapper = mount(Alert, {
      propsData: {
        show: true,
        closable: false,
      },
      attrs: {
        class: 'testing',
      }
    });

    expect(wrapper.find('.testing').exists()).toEqual(true);
    expect(wrapper.classes('testing')).toEqual(true);
  });

  it('should render title', () => {
    const wrapper = mount(Alert, {
      propsData: {
        show: true,
        title: 'test',
        dismissBtnProps: {
          ariaLabel: 'Close'
        }
      },
    });

    expect(wrapper.find('.md-alert__title').text()).toEqual('test');
  });

  it('should render message', () => {
    const wrapper = mount(Alert, {
      propsData: {
        show: true,
        message: 'test',
        dismissBtnProps: {
          ariaLabel: 'Close'
        }
      },
    });

    expect(wrapper.find('.md-alert__message').text()).toEqual('test');
  });

  it('should not display closable button if set to false', () => {
    const wrapper = mount(Alert, {
      propsData: {
        show: true,
        closable: false,
      },
    });

    expect(wrapper.find('.md-button').exists()).toEqual(false);
  });

  it('should display closable button by default', () => {
    const wrapper = mount(Alert, {
      propsData: {
        show: true,
        dismissBtnProps: {
          ariaLabel: 'Close'
        }
      },
    });

    expect(wrapper.find('.md-button').exists()).toEqual(true);
  });

  it('should pass type attribute props (success)', () => {
    const wrapper = mount(Alert, {
      propsData: {
        show: true,
        closable: false,
        type: 'success',
      },
    });

    expect(wrapper.find('.md-alert').classes('md-alert--success')).toEqual(true);
  });

  it('should pass type attribute props (warning)', () => {
    const wrapper = mount(Alert, {
      propsData: {
        show: true,
        closable: false,
        type: 'warning',
      },
    });

    expect(wrapper.find('.md-alert').classes('md-alert--warning')).toEqual(true);
  });

  it('should pass type attribute props (error)', () => {
    const wrapper = mount(Alert, {
      propsData: {
        show: true,
        closable: false,
        type: 'error',
      },
    });

    expect(wrapper.find('.md-alert').classes('md-alert--error')).toEqual(true);
  });

  it('should handle hide event', () => {
    let count = 0;
    const countUp = () => count++;
    const wrapper = mount(Alert, {
      propsData: {
        show: true,
        dismissBtnProps: {
          ariaLabel: 'Close'
        },
        dismissBtnEvents: {
          click: () => countUp()
        }
      },
    });

    wrapper.find('.md-button').trigger('click');
    expect(count).toEqual(1);
  });

  it('should pass dismissBtnProps prop', () => {
    const wrapper = mount(Alert, {
      propsData: {
        show: true,
        dismissBtnProps: {
          ariaLabel: 'Close'
        },
      },
    });

    expect(wrapper.find('button').attributes('aria-label')).toEqual('Close');
  });

});

