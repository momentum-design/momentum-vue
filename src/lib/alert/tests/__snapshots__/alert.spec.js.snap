// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Alert should match snapshot 1`] = `
<div class="md-alert md-alert--info">
  <div class="md-alert__icon"></div>
  <div class="md-alert__content">
    <div class="md-alert__title">

    </div>
    <div class="md-alert__message">

    </div>
  </div>
  <div class="md-alert__button"><button aria-label="Close" alt="Close" type="button" tabindex="0" role="button" class="md-button md-button--circle md-button--44"><span class="md-button__children" style="opacity: 1;"><i class="md-icon icon icon-cancel_16" style="font-size: 16px;"></i></span></button></div>
</div>
`;
